src/uct/cuda/cuda_copy/cuda_copy_iface.c

```diff
@@ -23,6 +23,15 @@
 #define UCT_CUDA_COPY_IFACE_LATENCY  ucs_linear_func_make(8e-6, 0)
 
 
+typedef struct uct_cuda_copy_ctx_mpool_priv {
+#if CUDA_VERSION >= 12000
+    CUcontext validator;
+#else
+    CUdeviceptr validator;
+#endif
+} uct_cuda_copy_ctx_mpool_priv_t;
```

Comment:
move validator to ctx_rsc
we can use ucs_container_of(mp, uct_cuda_copy_ctx_rsc_t, even_desc_mp) to get the ctx from obj_destroy